lnwallet/channel.go

```diff
@@ -3866,6 +3991,12 @@ func (lc *LightningChannel) AddHTLC(htlc *lnwire.UpdateAddHTLC) (uint64, error)
 		OnionBlob: htlc.OnionBlob[:],
 	}
 
+	remoteACKedIndex := lc.localCommitChain.tail().theirMessageIndex
+	if err := lc.validateCommitmentSanity(remoteACKedIndex,
```

Comment:
changing this to `remoteACKedIndex` makes the tests fail, but I believe this should be correct (and tests will need to be updated), as that is the index we use when `SignNextCommitment`? @Roasbeef

Reply:
Yes that's the index we use when we sign the next commit. Which tests fail?

Reply:
Oh actually, we use the log index as we want to ensure that if *either* party initiates an update, then the constraints aren't violated.

Reply:
Discussed offline, will ise acked index.

Reply:
See #485